src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorRecordServiceTest.java

```diff
@@ -3289,6 +3298,161 @@ private Appearance buildAppearance() {
+            assertEquals(BigDecimal.valueOf(6).setScale(2), payments.getSubsistence(), "Incorrect subsistence total");
+            assertEquals(BigDecimal.valueOf(6).setScale(2), payments.getTotalPaid(), "Incorrect total paid");
+            assertEquals(4, payments.getData().size(), "Incorrect number of attendance entries");
+        }
+
+        @Test
+        void negativeNoPermission() {
+            String jurorNumber = "641500094";
+            JurorPool jurorPool = createValidJurorPool(jurorNumber, COURT_OWNER);
+            TestUtils.mockSecurityUtil(
+                BureauJwtPayload.builder()
+                    .owner("416")
+                    .build()
+            );
+
+            doReturn(List.of(jurorPool)).when(jurorPoolRepository)
+                .findByJurorJurorNumberAndIsActive(any(), anyBoolean());
+
+            assertThrows(MojException.Forbidden.class, () -> jurorRecordService.getJurorPayments(jurorNumber));
+        }
+    }
+
+    @Nested
+    @DisplayName("public JurorHistoryResponseDto getJurorHistory(String jurorNumber)")
+    class GetJurorHistory {
+        String poolNumber = "1234567";
+
+        JurorHistory generateData(String item, Integer dateOffset) {
+            JurorHistory historyItem = mock(JurorHistory.class);
+            HistoryCodeMod code = mock(HistoryCodeMod.class);
+
+            doReturn(item).when(code).getDescription();
+
+            doReturn(code).when(historyItem).getHistoryCode();
+            doReturn(LocalDateTime.now().minusDays(dateOffset)).when(historyItem).getDateCreated();
+            doReturn(LocalDate.now().minusDays(dateOffset)).when(historyItem).getOtherInformationDate();
+            doReturn(poolNumber).when(historyItem).getPoolNumber();
+            doReturn(item).when(historyItem).getOtherInformation();
+            doReturn(String.valueOf(item.charAt(0))).when(historyItem).getOtherInformationRef();
+            doReturn("USER").when(historyItem).getCreatedBy();
+
+            return historyItem;
+        }
+
+        @AfterEach
+        void afterEach() {
+            TestUtils.afterAll();
+        }
+
+        @Test
+        void positiveGetHistory() {
+            String jurorNumber = "641500094";
+            JurorPool jurorPool = createValidJurorPool(jurorNumber, COURT_OWNER);
+            TestUtils.mockSecurityUtil(
+                BureauJwtPayload.builder()
+                    .owner("415")
+                    .build()
+            );
+
+            doReturn(List.of(jurorPool)).when(jurorPoolRepository)
+                .findByJurorJurorNumberAndIsActive(any(), anyBoolean());
+
+            doReturn(List.of(generateData("First", 1), generateData("Second", 2)))
+                .when(jurorHistoryRepository).findByJurorNumber(jurorNumber);
+
+            JurorHistoryResponseDto history = jurorRecordService.getJurorHistory(jurorNumber);
+
+            assertEquals(2, history.data.size(), "Incorrect history length");
```

Comment:
is the text here correct?  its meant to be a positive test

Reply:
I thought this was the error message if the assert failed? Maybe I've misunderstood.

Reply:
ok, we normally put a message like history length X expected but this also works I guess :-)